core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -946,6 +957,26 @@ public boolean containsUserDefinedFunction() {
     }
 
   }
+  /**
+   * Visitor for checking whether part of filter contains dynamic param
+   */
+  private static class CheckingDynamicParamVisitor extends RexVisitorImpl<Void> {
+
+    private boolean containsDynamicParam = false;
+
+    CheckingDynamicParamVisitor() {
+      super(true);
+    }
+
+    public boolean containsDynamicParam() {
+      return containsDynamicParam;
+    }
+
+    @Override public Void visitDynamicParam(RexDynamicParam dynamicParam) {
+      containsDynamicParam |= true;
```

Comment:
Is this intentional? You are forcing the value to 'true'
Maybe a simple assignment is clearer

Reply:
It is not intentional - I can simplify this.